src/System.Security.Cryptography.Csp/tests/PasswordDeriveBytesTests.cs

```diff
@@ -243,36 +243,77 @@ public static void GetBytes_IdempotentIfReset()
         }
 
         [Fact]
-        public static void GetBytes_StreamLike()
+        public static void GetBytes_StreamLike_ExtraBytes()
         {
             byte[] first;
 
             using (var deriveBytes = new PasswordDeriveBytes(TestPassword, s_testSalt))
             {
-                first = deriveBytes.GetBytes(32);
+                // SHA1 default
+                Assert.Equal("SHA1", deriveBytes.HashName);
+
+                // Request double of SHA1 hash size
+                first = deriveBytes.GetBytes(40);
             }
 
             byte[] second = new byte[first.Length];
 
             // Reset
             using (var deriveBytes = new PasswordDeriveBytes(TestPassword, s_testSalt))
             {
+                // Make two passes over the hash
                 byte[] secondFirstHalf = deriveBytes.GetBytes(first.Length / 2);
                 byte[] secondSecondHalf = deriveBytes.GetBytes(first.Length - secondFirstHalf.Length);
 
                 Buffer.BlockCopy(secondFirstHalf, 0, second, 0, secondFirstHalf.Length);
                 Buffer.BlockCopy(secondSecondHalf, 0, second, secondFirstHalf.Length, secondSecondHalf.Length);
             }
 
-            // Due to "_extraCount" bug in GetBytes() these won't be equal. The bug is fixed in Rfc2898DeriveBytes.
+            // Since we requested >= 20 bytes in secondFirstHalf there no 'extra' bytes and we don't encounter the
```

Comment:
I did tweak the comment. It is >= 20 bytes for secondFirstHalf that prevents the extra bytes and the fact the that the second call for the remainder gets all of the remaining.

For example if we call GetBytes(22) followed by GetBytes(18) that would go into the `if (ib >= cb)` (ib = 18; cb = 18) block which doesn't have the bug.

So it's not necessarily a mod 20 that prevents the bug although mod 40 total would be close (however the second call could ask for more than 40 total but the extra bytes would be ignored as we only do two passes here)